### PR TITLE
Add event_filter to salt cloud

### PR DESCRIPTION
**Pull request by techhat**
### What does this PR do?
See `doc/topics/cloud/reactor.rst` in the pull request for information.

### What issues does this PR fix or reference?
#39982

### Tests written?
No.

### CODE REVIEW

File: salt/cloud/clouds/dimensiondata.py [hunk 1]
```diff
@@ -164,42 +160,55 @@ def create(vm_):
...
-        kwargs = {
-            'name': vm_['name'],
-            'image': image,
-            'auth': rootPw,
-            'ex_description': vm_['description'],
-            'ex_network_domain': network_domain,
-            'ex_vlan': vlan,
-            'ex_is_started': vm_['is_started']
-        }
+    try:
```
Comment (tonybaloney) [inline salt/cloud/clouds/dimensiondata.py]: We can put this formatting in a separate PR if you want?

**Review by tonybaloney — COMMENTED** [review]
Love the functionality, but it's a shame that it means you have to update each and every driver.

**Comment by techhat** [comment]
Thanks @tonybaloney. I should elaborate.

Some of the drivers were not properly set up with either the `creating` or the `requesting` tag, so I made the appropriate changes. This was one of the drivers.

`__utils__['cloud.bootstrap']()` was in part designed to move a lot of the work that was similar or identical across all of the drivers into one place. However, firing events (outside of the `deploying` event in `bootstrap()` is still handled by individual drivers. The `requesting` event in particular varied greatly. It would be nice to standardize those events so that we didn't have to do this kind of change for them, but a) that's a whole lot more work than I did here and b) this kind of mass change isn't really isn't really very common anymore.

I should also note that many of the events for some reason were passing off the `driver` as the `provider`, which is obviously incorrect. This change fixes that, and presents both data in the event (where appropriate).

**Comment by techhat** [comment]
Go Go Jenkins!

**Review by gtmanfred — APPROVED** [review]

**Comment by rallytime** [comment]
@techhat I like this idea! 

Looks like some lint needs to be fixed up, though. :)

https://jenkins.saltstack.com/job/PR/job/salt-pr-lint-n/9466/violations/

**Comment by techhat** [comment]
@rallytime was already working on them. Should be cleared up now.

**Review by rallytime — APPROVED** [review]
